collector/nvme_linux.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+// +build !nonvme
+
+package collector
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs/sysfs"
+)
+
+type nvmeCollector struct {
+	fs        sysfs.FS
+	logger    log.Logger
+	subsystem string
+}
+
+func init() {
+	registerCollector("nvme", defaultEnabled, NewNVMeCollector)
+}
+
+// NewNVMeCollector returns a new Collector exposing NVMe stats.
+func NewNVMeCollector(logger log.Logger) (Collector, error) {
+	var i nvmeCollector
```

Comment:
nit: let's do this instead to be more idiomatic and consistent with other collector
```
	fs, err := procfs.NewFS(*procPath)
	if err != nil {
		return nil, fmt.Errorf("failed to open procfs: %w", err)
	}
        return &nvmeCollector{
        	fs: fs,
        	logger: logger,
        	subsystem: "nvme"
        }
```

Reply:
Changed and moved `subsystem` out to become a constant instead.